### DISCUSSION TITLE
Get Extension version number

### DISCUSSION THREAD

**Discussion by aghayanlena** [eclipse-theia/theia#14135] [General] [answered]
Hi , 

Could you please guide me on how I can get the version of the custom plugin by name? 

Thanks.

**Comment by msujew** [comment]
Hey @aghayanlena,

you can inject the `HostedPluginSupport` service in your frontend and use the `getPlugin` to get the plugin metadata. The metadata contains the version as specified in the package.json of the plugin.

**Comment by aghayanlena** [comment]
Thanks for the details. 

How can I get the PluginIdentifiers.UnversionedId  based on the extension name  to pass to getPlugin ?

**Reply by msujew** [reply, marked as answer]
The `UnversionedId` is the combined string of `<publisher>.<name>` from the plugin manifest (`package.json`). See also `PluginIdentifiers.componentsToUnversionedId`

**Reply by aghayanlena** [reply]
Thanks, it worked.
